crates/cargo-wdk/src/cli/args.rs

```diff
@@ -0,0 +1,174 @@
+                s.to_string(),
+                InvalidDriverProjectNameError::EmptyProjectNameError,
+            ));
+        }
+        if !s
+            .chars()
+            .all(|c| c.is_alphanumeric() || c == '-' || c == '_')
+        {
+            return Err(NewProjectArgsError::InvalidDriverProjectNameError(
+                s.to_string(),
+                InvalidDriverProjectNameError::NonAlphanumericProjectNameError,
+            ));
+        }
+        if !s
+            .chars()
+            .next()
+            .expect("Project name cannot be empty")
+            .is_alphabetic()
+        {
+            return Err(NewProjectArgsError::InvalidDriverProjectNameError(
+                s.to_string(),
+                InvalidDriverProjectNameError::InvalidStartCharacter,
+            ));
+        }
+        let invalid_names = ["crate", "self", "super", "extern", "_", "-", "new", "build"];
+        if invalid_names.contains(&s) {
+            return Err(NewProjectArgsError::InvalidDriverProjectNameError(
+                s.to_string(),
+                InvalidDriverProjectNameError::ReservedName(s.to_string()),
+            ));
+        }
+        std::result::Result::Ok(Self(s.to_string()))
+    }
+}
+
+/// Type for Driver Type Argument
+#[derive(Debug, Clone)]
+pub enum DriverTypeArg {
+    Kmdf,
+    Umdf,
+    Wdm,
+}
+
+impl From<DriverTypeArg> for DriverType {
+    fn from(val: DriverTypeArg) -> Self {
+        match val {
+            DriverTypeArg::Kmdf => Self::Kmdf,
+            DriverTypeArg::Umdf => Self::Umdf,
+            DriverTypeArg::Wdm => Self::Wdm,
+        }
+    }
+}
+
+impl FromStr for DriverTypeArg {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "kmdf" => std::result::Result::Ok(Self::Kmdf),
+            "umdf" => std::result::Result::Ok(Self::Umdf),
+            "wdm" => std::result::Result::Ok(Self::Wdm),
+            _ => Err(NewProjectArgsError::InvalidDriverTypeError(s.to_string()).to_string()),
+        }
+    }
+}
+
+/// Arguments for the new project subcommand
+/// This struct is used to parse the command line arguments for creating a new
+/// driver project.
+#[derive(Debug, Args)]
+pub struct NewProjectArgs {
+    #[clap(help = "Driver Project Name")]
+    pub driver_project_name: ProjectNameArg,
+    #[clap(long, help = "Driver Type", index = 2, ignore_case = true)]
+    pub driver_type: DriverTypeArg,
+    #[clap(long, help = "Path to the project", default_value = ".")]
+    pub cwd: PathBuf,
+}
+
+/// Type for Profile Argument
+#[derive(Debug, Clone)]
+pub enum ProfileArg {
+    Debug,
+    Release,
+}
+
+impl From<ProfileArg> for Profile {
+    fn from(val: ProfileArg) -> Self {
+        match val {
+            ProfileArg::Debug => Self::Debug,
+            ProfileArg::Release => Self::Release,
+        }
+    }
+}
+
+impl FromStr for ProfileArg {
```

Comment:
Same here. `Arg` suffix it not needed.

Reply:
https://github.com/microsoft/windows-drivers-rs/pull/306#discussion_r1981101232